common/utils.js

```diff
@@ -1794,14 +1806,39 @@
         function setConfigJSON(catalogAnnotation) {
             var cc = {};
             // check to see if global chaise-config (chaise-config.js) is available
-            if (typeof chaiseConfig != 'undefined') cc = Object.assign({}, chaiseConfig);
+            if (typeof chaiseConfig != 'undefined') {
+                // loop through properties and compare to defaultConfig to see if they are valid
+                // chaiseConfigPropertyNames is a whitelist of all accepted values
+                for (var key in chaiseConfig) {
+                    // see if returned key is in the list we accept
+                    var matchedKey = chaiseConfigPropertyNames.filter(function (defaultKey) {
```

Comment:
You've already done in multiple places, but you could define a private helper function in this factory that does this for you (so you don't have to copy/paste the code)